src/dbnode/integration/setup.go

```diff
@@ -186,14 +186,15 @@ type TestSetup interface {
 	InitializeBootstrappers(opts InitializeBootstrappersOptions) error
 }
 
-type storageOption func(storage.Options) storage.Options
+// StorageOption is a reference to storage options function.
+type StorageOption func(storage.Options) storage.Options
 
 // NewTestSetup returns a new test setup for non-dockerized integration tests.
 func NewTestSetup(
 	t *testing.T,
 	opts TestOptions,
 	fsOpts fs.Options,
-	storageOptFns ...storageOption,
+	storageOptFns ...StorageOption,
 ) (TestSetup, error) {
```

Comment:
There are two sources of `StorageOption` in this function: the `storageOptFns` varargs and `StorageOptsFn()` in `TestOptions`. I think it might cause some confusion (e.g. `StorageOption` inside `TestOptions` would get applied if it's passed to `NewDefaultBootstrappableTestSetups()` method, but not if passed directly to `NewTestSetup()`). Would it make sense leave only one of them? It seems that this `storageOptFns` vararg is only used in a handful of place, maybe we could get rid of it?

Reply:
That's right. This function is a little bit chaotic (for example multiple AdminClients) but I'd like to keep a bigger refactoring away from this PR.